BAHC4-121 add basic styles

MigleMat opened the pull request:
Hello @alex-koshara! My homework is ready for your review. 

First of all, I would like to say that I didn't really know how to work neatly and organized with my classes. Maybe any advise here? 
I didn't know if first I need to style headings or just go step by steps in order. So my style sheet, I think, look a bi messy :( 

**Buttons**  :D 
My secondary button is smaller than primary button. The size should be the same. Is it because I put a border? How to fix it? 
![Screenshot 2025-03-03 181230](https://github.com/user-attachments/assets/0e92a8a1-c422-4919-b701-49797febeb73)
![Screenshot 2025-03-03 181509](https://github.com/user-attachments/assets/78f41145-00ee-48b9-afc5-20167dbe8f34)

I met few times in my work, where I have example like this: I have h1 in my home page and blog page. Both of them have the same values, except one - text align. So in this case. Do I repeat the same values in each class? or (as I did now), first I target tag selector h1 and listet all values, which are same for both and after I target these headings seperate classes and listed their different values. In preview worked, but I am a bit confused here. 

![Screenshot 2025-03-03 181647](https://github.com/user-attachments/assets/b078f717-c4bc-4fc9-8015-ce7a66c50489)
![Screenshot 2025-03-03 181712](https://github.com/user-attachments/assets/3906a7f3-d2a8-4cde-9f0f-5b903b861d86)

If I need to choose, which selector to target tag or class, because that tag doesn't repeat nowhere else, it means that I don't need class at all, right? :)  I have only in one place h3.. 

Why do I have this white line between header main and footer? 
![Screenshot 2025-03-03 182514](https://github.com/user-attachments/assets/e4d64630-059a-4a13-accc-c7330e1297b6)

What's up with my arrow buttons? They have same classes as others, but they don't act same way. Why??? I couldn't find answer myself :( 

![Screenshot 2025-03-03 182600](https://github.com/user-attachments/assets/a26abc57-a798-4a7c-bc19-9082259ffb9e)
![Screenshot 2025-03-03 182733](https://github.com/user-attachments/assets/3796e636-ba5e-469c-a90a-f61173d09c8a)

In my home page and blog page I have _article-writer and date_, which are the same font style, however they have different classes. So I repeat same style for both pages, but I think I shouldn't. Right? Do I need a seperate block for them, so can give style once for both pages? 

![Screenshot 2025-03-03 183112](https://github.com/user-attachments/assets/953c8eee-86df-4096-9bd4-920e37343b0d)
![Screenshot 2025-03-03 183404](https://github.com/user-attachments/assets/b362da92-4457-4653-b9ba-e6b8f73dc28e)

Also, I was confused what to do with these numbers. But as I remember, teacher said that this wll be done in futures another classes? maybe I am working, please advise. 
![Screenshot 2025-03-03 183525](https://github.com/user-attachments/assets/b645573d-1bef-4e83-b386-df1736554b49)
![Screenshot 2025-03-03 183556](https://github.com/user-attachments/assets/da584959-e425-4b0a-874c-6d9f64c1408c)


@alex-koshara I will wait for your reply and other your observations, of course. Thank you in advance for taking care of it.)))

MigleMat commented:
Also, can you please review and comment about my google fonts and normalize.css ?

alex-koshara commented on blog.html:
```diff
@@ -3,11 +3,18 @@
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Bakbak+One&family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap" rel="stylesheet">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Bakbak+One&family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap" rel="stylesheet"> 
```
Could you please clarify, why these rows repeats?

MigleMat replied:
My mistakes. Going to delete one.

alex-koshara commented on blog.html:
```diff
@@ -3,11 +3,18 @@
-    <link rel="stylesheet" href="styles.css" />
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Bakbak+One&family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap" rel="stylesheet">
```
Could you please clarify, do you need all these fonts?

alex-koshara commented on index.html:
```diff
@@ -3,11 +3,18 @@
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Bakbak+One&family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="style.css" />
+    <link rel="stylesheet" type="text/css" href="https://necolas.github.io/normalize.css/8.0.1/normalize.css"/>
```
Please tell me, are you sure that normalize needs to be enabled after your main styles?

MigleMat replied:
Fixed!

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+body {
+  font-family: Inter;
```
Everything related to fonts is inherited. That is, if we set a font for body, all child elements will have exactly the same font. It also works for font size, color, line-height and etc.

Look at which font (color and size) is most often found, and I suggest adding it for the body, so that later it is less prescribed in classes.

alex-koshara replied:
Could you please clarify, where you found `Inter` font in figma? It looks like we don't need it

MigleMat replied:
@alex-koshara so, I downloaded my google font and, I don't know how, I thought _Bakbak+One_ fonts are different and I need them both, and that _Inter_ font clasifies them both ![image](https://github.com/user-attachments/assets/e7f75d5b-c039-4fc4-af16-ec1aba84d036) now I understand it's not, it was silly mistakes ![image](https://github.com/user-attachments/assets/43f8f50c-334d-48b1-9f66-e40e65f522c8)
 
![Screenshot 2025-03-04 073234](https://github.com/user-attachments/assets/d0739f70-579e-408d-a857-b674da8c54a0)

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+}
+
+.header {
+  background-color: rgb(13, 13, 21, 1);
+  color: rgb(245, 248, 251); 
```
Could you please describe why we use here `rgb` and not `hex` color?
I also suggest putting the [colors in css variables](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_cascading_variables/Using_CSS_custom_properties)

MigleMat replied:
hm.... Isn't my choise here to use whatever I want? I asked this during the lecture and teacher said it can be any (I might be wrong, maybe missheard something)

alex-koshara replied:
If color don't have alpha channel, not need to use it, just only rgb or hex

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+  background-color: rgb(13, 13, 21, 1);
+  color: rgb(245, 248, 251); 
+}
+
+ul {
+  list-style-type: none;
+}
+.nav__link {
```
One of the important qualities in development is to follow uniformity on the project. If we have spaces between selectors, they should be everywhere. There is an empty string before the ul, but not after it. Let's add a line here, too, for uniformity, and if it repeats somewhere, then we'll add it too for uniformity.

MigleMat replied:
noted, thanks

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+}
+.nav__link {
+color: rgb(245, 248, 251);
+font-family: Poppins;
```
We also can add font-name to variable. And please  add also alternative font and family

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+font-size: 16px;
+font-weight: 600;
+line-height: 24px;
+letter-spacing: 0px;
```
One important point. Zeros have no units of measurement. Zero is always zero. Please delete the `px` here and if there is somewhere else too

MigleMat replied:
fixed!

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+}
+
+.nav__link--selected {
+  border-bottom: 2px solid rgba(2, 101, 220, 1); 
```
Could you please clarify, why here in code we have `rgba`, but  in head `rgb` for color
<img width="343" alt="image" src="https://github.com/user-attachments/assets/c8199644-cb93-48b5-bb2d-6bd884ee335f" />

MigleMat replied:
@alex-koshara  cause in some values from pixso, I couldnt find opacity to clarify in values, so I used just rgb. But in all my work format is consistent, I used rgba or rgb. Not like hex and rgb. Is it wrong?

MigleMat replied:
oh, in this your example, I see that I am wrong. it has A.. going to fix that. But in some places in my work I do have examples like I mention before. Is it wrong?

alex-koshara replied:
It's not an error. But you need to chose one way and use it whole project 🙌

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+  border-bottom: 2px solid rgba(2, 101, 220, 1); 
+}
+
+.button {
+color: rgb(245, 248, 251);
+font-family: Poppins;
+box-sizing: border-box;
+border: 2px solid rgba(208, 215, 251, 0.17);
+border-radius: 0;
+font-weight: 600;
+font-size: 18px;
+}
```
Please add spaces for rows 28-34

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+.button {
+color: rgb(245, 248, 251);
+font-family: Poppins;
+box-sizing: border-box;
```
How about add it one time for each selector?

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+  line-height: 28px;
+  letter-spacing: 0px;
+}
+a {
+  text-decoration: none;
+}
```
I would recommend resetting styles as needed in the future. For example, we may have a text that comes through the admin panel, and sometimes it is necessary that there be a standard behavior for links. But for this project, let's leave it, let's see what happens

MigleMat replied:
I will keep that in mind!

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+  background-color: rgb(13, 13, 21, 1);
+}
+
+h1 {
```
Please add these styles to class, not to tag

MigleMat replied:
fixed!

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+text-align: left;
+}
+
+h2 {
```
We have a criterion that says that we should not use selectors by tags. How about we replace all such selectors (except those allowed in the criteria) with classes?

MigleMat replied:
done

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+}
+
+.label--updated {
+
```
Please delete empty row

MigleMat replied:
done

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+} 
+
+.social {
+  display: flex;
```
This line not from this task 😈

MigleMat replied:
sorry, I was just playing to see, how it will look like (I was just curious) and I forgot to delete.  Deleted

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+
+}
+
+footer {
```
Please use class, not tag

MigleMat replied:
done

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,501 @@
+}
+
+.social__list {
+  display: flex;
```
Looks like we already have flex for `social` 🤔

MigleMat replied:
deleted

alex-koshara reviewed (commented):
Pretty good job! The assignment was understood correctly. Fonts, colors, line-height and other properties for typography are prescribed. 

Let's correct the general comments now.:
- let's remove the styling by tags
- remove the unit of measurement for zeros
- add variables for colors and fonts
- remove the unused font Inter

And I also want to suggest installing the prettier plugin, it should help with the alignment of the css code. Alternatively, you can try to look for any other plugin for formatting styles.

alex-koshara commented:
> First of all, I would like to say that I didn't really know how to work neatly and organized with my classes. Maybe any advise here?
> I didn't know if first I need to style headings or just go step by steps in order. So my style sheet, I think, look a bi messy :(

Not yet, we'll do it when we'll learn pre-processor (SCSS)


About button:
> The size should be the same. Is it because I put a border? How to fix it?

Yes, try to remember about box model. Or try to do the same with primary button


About H1:
>Do I repeat the same values in each class?

Now yes, we will repeat the styles. There are several patterns and each has its pros and cons. I understand that this increases the number of lines of code, but the most important thing is that it looks uniform now.

> it means that I don't need class at all, right?

No, we always use only classes for styling 😸 

>I was confused what to do with these numbers.

Just skip it now, it will be in the next lesson 🙌 


I suggest we sort out the rest of the questions at the consultation to see how you tried to find a solution, and then I'll show you how to look for problems and solve them.

MigleMat commented:
Hey @alex-koshara! So I made the changes (hopefully correct): 
1. Removed the styling by tags (except a)
2. Removed the unit of measurement for zeros
3. Added variables for colors and fonts (please double check this one)
4. Remove the unused font Inter and fixed my google font in head 
5.So I fixed my colors and used evreywhere rgba. I hope it's correct?

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,721 @@
+
+}
+
+body {
+
+  font-family: var(--main-font);
+  color: var(--primary-color);
+
+}
```
Is this style did prettier or yourself? One line before bracket and after

MigleMat replied:
Much better! 🫠

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,721 @@
+:root {
+
+  --primary-color: rgba(248, 247, 255, 1);
+  --main-font: 'Bakbak One','Poppins', Helvetica, Arial, sans-serif;
```
Nice work! You also can add secondary color and secondary font if you want!

And let a little fix it. We need only one main font here, one font alternative and one family.

Poppins font looks like secondary font for project. What do you think?

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,721 @@
+.nav__link {
+
+  color: rgba(245, 248, 251, 1);
+  font-family: 'Poppins', sans-serif;
```
Please add also some alternative font

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,721 @@
+  font-size: 16px;
+  font-weight: 600;
+  line-height: 24px;
+  letter-spacing: 0;
```
Do we need letter spacing if it's 0 by default?

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,721 @@
+  text-align: left;
+  text-transform: uppercase;
+  color: rgba(245, 248, 251, 1);
+  font-family: 'Bakbak One', sans-serif;
```
Please use variable 😸

alex-koshara commented on style.css:
```diff
@@ -0,0 +1,721 @@
+}
+
+
+/* .bakbak-one-regular {
```
Please delete comments if it not need now

MigleMat replied:
Will delete after consultation, I have a question about it

alex-koshara reviewed (commented):
Very good! There's just a little bit left and a couple of small details!

MigleMat commented:
Hello, @alex-koshara! So I made all the changes in my CSS document, I think it looks pretty good, at least visually after I used prettier extention :smile: I believe I understood all my mistakes, memorized and corrected them. 💪 
Just I left everywhere oppacity as we discussed, but I will keep that in mind for next time. 
Also, I still have one question (I wrote it in slack). Unexpected error appeared after validation. Let me know if I should do something about it, cause it didn't change my preview. 

Thank you so much for your enthusiasm for teaching!
